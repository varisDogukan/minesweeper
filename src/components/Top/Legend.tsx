import React from "react";
import styled from "@emotion/styled";

export interface LegendProps {
  feature: string;
  firstAction: string;
}

export default function Legend({ feature, firstAction }: LegendProps) {
  return (
    <Parent>
      <strong>{feature}: </strong>
      <FlagComboParent>
        <FirstAction>{firstAction}</FirstAction>
      </FlagComboParent>
    </Parent>
  );
}

const FlagComboParent = styled.code`
  background-color: #e3e3e3;
`;

const Parent = styled.legend`
  font-size: 1em;
  margin: 0 auto 2vw;
  line-height: 1.25em;
`;

const FirstAction = styled.span`
  color: #ec433c;
`;
