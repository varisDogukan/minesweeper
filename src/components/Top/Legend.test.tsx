import React from "react";
import { render } from "@testing-library/react";

import Legend from "./Legend";

it("GameName renders correctyle", () => {
  const { asFragment } = render(
    <Legend feature='Flag' firstAction='Ctrl' secondAction='Click' />
  );

  expect(asFragment()).toMatchSnapshot();
});
