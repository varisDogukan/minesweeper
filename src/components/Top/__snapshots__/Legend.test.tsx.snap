// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameName renders correctyle 1`] = `
<DocumentFragment>
  <legend
    class="css-17qz6a2"
  >
    <strong>
      Flag: 
    </strong>
    <code
      class="css-h9xasx"
    >
      <span
        class="css-udfepp"
      >
        Ctrl
      </span>
       + 
      <span
        class="css-9c23k1"
      >
        Click
      </span>
    </code>
  </legend>
</DocumentFragment>
`;
